src/settings.rs

```diff
@@ -15,120 +16,86 @@ lazy_static! {
     pub static ref SETTINGS: Settings = Settings::new();
 }
 
-#[derive(Debug)]
-pub enum Setting {
-    Bool(bool),
-    U16(u16),
-    String(String)
+struct SettingsObject {
```

Comment:
A high level comment at the top of this file somewhere describing the plan and reasoning would help make this understandable for anyone trying to puzzle through the settings architecture in the future.